Add dnssd support to cpu (#6)

ericvh opened the pull request:
Add a new module (ds) which can be used to register and discover cpu servers with dnssd.  Create seperate cmds for cpu and cpud (decpu and decpud) which incorporate the use of dnssd while still using same base client and server modules as original.

ds uses meta-data in the dns TXT region to relate information about the host decpud is running on (including GOARCH, GOOS, as well as core count, memory information, load information, and number of cpu clients currently connected.  This information can be augmented from the decpud command line to include other meta-data (such as near-NIC, near-GW, near-storage, near-GPU, near-robot, etc.)

Lookup is specified as a URI (with sensible defaults applied) which is used in place of a hostname and can be omitted to take advantage of defaults.  meta-data requirements and sorting can be specified as query parameters to the URI.  e.g. dnsrd://?arch=arm64&sort=load

Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>

ericvh committed:
Add dnssd support to cpu (#6)

Add a new module (ds) which can be used to register and discover cpu
servers with dnssd.  Create seperate cmds for cpu and cpud (decpu
and decpud) which incorporate the use of dnssd while still using same
base client and server modules as original.

ds uses meta-data in the dns TXT region to relate information about the
host decpud is running on (including GOARCH, GOOS, as well as core
count, memory information, load information, and number of cpu clients
currently connected.  This information can be augmented from the
decpud command line to include other meta-data (such as near-NIC,
near-GW, near-storage, near-GPU, near-robot, etc.)

Lookup is specified as a URI (with sensible defaults applied) which is
used in place of a hostname and can be omitted to take advantage of
defaults.  meta-data requirements and sorting can be specified as
query parameters to the URI.  e.g. dnsrd://?arch=arm64&sort=load

Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>

rminnich reviewed (approved):
not a fan of the name but it's no worse than the name cpu, so ... I don't know why I can't have a rebase button but ... you'll have to update it and git push -f ...

ericvh commented:
> not a fan of the name but it's no worse than the name cpu, so ... I don't know why I can't have a rebase button but ... you'll have to update it and git push -f ...

rebased.

ericvh committed:
add some basic documentation for decent-e variants

Addition to the docs which describe using decpu and decpud
as well as what we need to do in order to use them under docker on Mac.

Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>

ericvh committed:
Make it so default requirements can be ignored

Realized that with default requirements, there was no way to tell cpu to
ignore them, just provide alternate required values.  Use '*' as a
value to make keys that can be ignored.

Signed-off-by: Eric Van Hensbergen <ericvh@gmail.com>
